packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -120,38 +78,80 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
     };
 
     try {
-      await this.webSocket.subscribeAndConnect({
-        datachannelUrl,
-        body: connectionConfig,
-      });
-      this.$webex.logger.info('Successfully connected and subscribed.');
+      return this.httpRequest
+        .subscribeNotifications({
+          body: connectionConfig,
+        })
+        .then(async (data: WelcomeEvent) => {
+          const agentId = data.agentId;
+          const agentConfig = new AgentConfig(agentId, this.$webex, this.httpRequest);
+          this.agentConfig = await agentConfig.getAgentProfile();
+          this.$webex.logger.log(`file: ${CC_FILE}: agent config is fetched successfully`);
+
+          return this.agentConfig;
+        })
+        .catch((error) => {
+          throw error;
+        });
     } catch (error) {
-      this.$webex.logger.info(`Error connecting and subscribing: ${error}`);
-      reject(error);
+      this.$webex.logger.error(`file: ${CC_FILE}: Error during register: ${error}`);
+
+      throw error;
     }
   }
 
-  private handleEvent(eventName: string, result: EventResult) {
-    const handler = this.eventHandlers.get(eventName);
-    if (handler) {
-      clearTimeout(handler.timeoutId);
-      handler.resolve(result);
-      this.eventHandlers.delete(eventName);
+  /**
+   * This is used for agent login.
+   * @param options
+   * @returns Promise<StationLoginSuccess>
+   * @throws Error
+   */
+  public async stationLogin(data: AgentLoginRequest): Promise<StationLoginResponse> {
+    try {
+      let callingSDKRegister: Promise<void> | null = null;
+
+      if (data.loginOption === LoginOption.BROWSER) {
+        this.webRTCCalling = new WebRTCCalling(this.$webex, {}); // TODO: add callingClientConfig
+        callingSDKRegister = this.webRTCCalling.registerWebCallingLine();
+        data.dialNumber = this.agentConfig.agentId; // replacing dialNumber with agentId for BROWSER case
+      }
+
+      const loginPromise = this.services.agent.stationLogin({
+        data: {
+          dialNumber: data.dialNumber || this.agentConfig.agentId,
+          teamId: data.teamId,
+          deviceType: data.loginOption,
+          isExtension: data.loginOption === LoginOption.EXTENSION,
+          deviceId: this.getDeviceId(data.loginOption, data.dialNumber),
+          roles: [AGENT],
+          teamName: '',
+          siteId: '',
+          usesOtherDN: false,
+          auxCodeId: '',
+        },
+      });
+
+      if (callingSDKRegister) {
+        // LoginOption.BROWSER case we have to wait until calling sdk also registered.
+        await Promise.all([callingSDKRegister, loginPromise]);
+      } else {
+        await loginPromise;
+      }
+
+      this.$webex.logger.log(`file: ${CC_FILE}: Station Login Success`);
+
+      return loginPromise;
+    } catch (error: any) {
+      this.$webex.logger.log(`file: ${CC_FILE}: Station Login FAILED: ${error.id}`);
```

Comment:
we need to makes sure specific errors are managed calling or login , add TODO to make it better